vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -555,24 +555,19 @@ - (void)onGrant:(ledger::Result)result grant:(ledger::GrantPtr)grant
 
 - (void)grantCaptchaForPromotionId:(NSString *)promoID promotionType:(NSString *)promotionType completion:(void (^)(NSString * _Nonnull, NSString * _Nonnull))completion
 {
-  const auto __weak weakSelf = self;
-  self.grantCaptchaBlock = ^(const std::string &image, const std::string &hint) {
-    weakSelf.grantCaptchaBlock = nil;
-    completion([NSString stringWithUTF8String:image.c_str()],
-               [NSString stringWithUTF8String:hint.c_str()]);
-  };
   std::vector<std::string> headers;
   headers.push_back("brave-product:brave-core");
   headers.push_back("promotion-id:" + std::string(promoID.UTF8String));
   headers.push_back("promotion-type:" + std::string(promotionType.UTF8String));
-  ledger->GetGrantCaptcha(headers);
+  ledger->GetGrantCaptcha(headers,
+      ^(const std::string &image, const std::string &hint) {
+        completion([NSString stringWithUTF8String:image.c_str()],
```

Comment:
Is grant captcha doing an asynchronous network call? I can't recall, but if it is, can we wrap this completion block in a `dispatch_async` to main thread as seen in other callbacks

Reply:
Fixed